python/lsst/source/injection/inject_base.py

```diff
@@ -0,0 +1,636 @@
+    mask_plane_name = Field[str](
+        doc="Name assigned to the injected mask plane which is attached to the output exposure.",
+        default="INJECTED",
+    )
+    calib_flux_radius = Field[float](
+        doc="Aperture radius (in pixels) that was used to define the calibration for this image+catalog. "
+        "This will be used to produce the correct instrumental fluxes within the radius. "
+        "This value should match that of the field defined in ``slot_CalibFlux_instFlux``.",
+        default=12.0,
+    )
+    fits_alignment = ChoiceField[str](
+        doc="How should injections from FITS files be aligned?",
+        dtype=str,
+        allowed={
+            "wcs": (
+                "Input image will be transformed such that the local WCS in the FITS header matches the "
+                "local WCS in the target image. I.e., North, East, and angular distances in the input image "
+                "will match North, East, and angular distances in the target image."
+            ),
+            "pixel": (
+                "Input image will **not** be transformed. Up, right, and pixel distances in the input image "
+                "will match up, right and pixel distances in the target image."
+            ),
+        },
+        default="pixel",
+    )
+    stamp_prefix = Field[str](
+        doc="String to prefix to the entries in the *col_stamp* column, for example, a directory path.",
+        default="",
+    )
+    draw_size_scale = Field[float](
+        doc="Multiplicative scale factor applied to the size of the injected box drawn for each source.",
+        default=1.0,
+    )
+    draw_size_max = Field[int](
+        doc="Maximum allowed pixel size of the injected box drawn for each source. "
+        "Clipped to this value if set greater than zero.",
+        default=10000,
+    )
+
+    # Custom column names.
+    col_ra = Field[str](
+        doc="Column name for right ascension (in degrees).",
+        default="ra",
+    )
+    col_dec = Field[str](
+        doc="Column name for declination (in degrees).",
+        default="dec",
+    )
+    col_mag = Field[str](
+        doc="Column name for magnitude.",
+        default="mag",
+    )
+    col_source_type = Field[str](
+        doc="Column name for the source type used in the input catalog. Must match one of the surface "
+        "brightness profiles defined by GalSim.",
+        default="source_type",
+    )
+    col_select = Field[str](
+        doc="Column name for boolean flags used to identify which sources to inject. "
+        "Will only be applied when *subset_by_select* is True.",
+        default="select",
+    )
+    col_stamp = Field[str](
+        doc="Column name to identify FITS file postage stamps for direct injection. The strings in this "
+        "column will be prefixed with a string given in *stamp_prefix*, to assist in providing the full "
+        "path to a FITS file.",
+        default="stamp",
+    )
+    col_draw_size = Field[str](
```

Comment:
This is a fairly important configuration option that was added in at request from several end-users. I worry that removing it will hamper our ability to pursue some of the things we want to do out of the gate until a more suitable approach comes up.

However, if we're happy to keep it in the short term, I'm happy to set up a ticket to look at replacing it?

Reply:
PS - I'm also happy to remove the (perhaps redundant) `draw_size_scale` config option as we discussed last week, if that would be appropriate here?

Reply:
On discussion, we've agreed to leave `col_draw_size` in place for now (the config option which sets the name of the draw size column, default "draw_size"). This functionality is likely something that we'll want out of the box. We will however remove `draw_size_scale` and `draw_size_max` from this PR, as those are arguably unnecessary extras.

Reply:
Both `draw_size_scale` and `draw_size_max` have been removed. The `col_draw_size` config remains, allowing the user to manually specify a draw size override.